### PR TITLE
Basic conversation to TS2 and 'grunt-dojo2'

### PR DESCRIPTION
**Pull request by agubler**
Should enable TS2.0 compilation and integrates grunt-dojo2 for the build pipeline.

Requires changes in PR https://github.com/dojo/grunt-dojo2/pull/26

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +34,6 @@
     "grunt-tslint": "^3.2.1",
     "intern": "^3.3.1",
     "tslint": "^3.15.1",
-    "typescript": "^2.1.0-dev.20160828"
+    "typescript": "2.1.0-dev.20160819"
```
Comment (agubler) [inline package.json]: pinned to the same version that dojo2 are pinned to.

File: Gruntfile.js [hunk 2]
```diff
@@ -255,100 +180,83 @@ module.exports = function (grunt) {
...
 		}
+	},
+	{
+		istanbulIgnoreNext: '/* istanbul ignore next */',
+		staticTestFiles: 'tests/**/*.{html,css}',
+		distDirectory: 'dist',
+		testDirectory: 'test',
+		srcDirectory: 'src',
+		targetDirectory: '<%= devDirectory %>',
+		siteDirectory: '.'
```
Comment (devpaul) [inline Gruntfile.js]: When running `grunt` I get an error:

```
Warning: An error occurred while processing a template (siteDirectory is not defined). Use --force to continue.
```

It doesn't look like there is a 3rd parameter in grunt-dojo2?

File: Gruntfile.js [hunk 3]
```diff
@@ -255,100 +180,83 @@ module.exports = function (grunt) {
...
-			grunt.verbose.writeln('Renamed ' + file.src[0] + ' to ' + file.dest);
-		});
-		grunt.log.writeln('Moved ' + this.files.length + ' files');
-	});
-
-	grunt.registerTask('updateTsconfig', function () {
-		var tsconfig = JSON.parse(tsconfigContent);
-		tsconfig.files = grunt.file.expand(tsconfig.filesGlob);
+					// Set some Intern-specific options if specified on the command line.
+					[ 'suites', 'functionalSuites', 'grep' ].forEach(function (option) {
```
Comment (devpaul) [inline Gruntfile.js]: looks like this is in grunt-dojo2 on master. Can we get rid of this?

Comment (agubler) [reply Gruntfile.js]: I think so, missed that.

File: Gruntfile.js (hunk 2 above)
Comment (agubler) [reply Gruntfile.js]: It required changes in grunt-dojo2, but I have closed that and replaced it with another proposed change.... As soon as that is in and released I'll update this PR.

**Comment by agubler** [comment]
@devpaul could you not get the project building using grunt-dojo2?

**Comment by agubler** [comment]
@devpaul assuming you don't want any of these suggested changes and am closing

**Comment by devpaul** [comment]
@agubler Missed your comment. I wasn't able to use `grunt-dojo2` because I couldn't figure out how to override grunt variables in the amount of time I had to try to fix it. I was also worried it might not be flexible enough and apparent what people needed to change for a boilerplate.

I added your PR in @ fc2cdc9147f4240c896d76a77fc30db66fc8efc4 I tweaked it a little, but it helped a lot (!) to get this all working under TS w/ Dojo 2. Thank you!

**Pull request closed without merging**